crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/service/RepOrderServiceTest.java

```diff
@@ -304,37 +254,133 @@ void givenIndictableCaseWithFailedAssessment_whenGetRepDecisionIsInvoked_granted
+                        CaseType.INDICTABLE,
+                        MagCourtOutcome.APPEAL_TO_CC,
+                        CurrentStatus.COMPLETE,
+                        null,
                         InitAssessmentResult.FAIL.getResult(),
                         null,
-                        null),
-                // Scenario 2: Failed initial assessment and failed full assessment.
-                Arguments.of(CurrentStatus.COMPLETE, CurrentStatus.COMPLETE,
+                        null,
+                        Constants.GRANTED_FAILED_MEANS_TEST
+                ),
+                /*
+                 * Scenario 2: INDICTABLE case with MagCourtOutcome of APPEAL_TO_CC.
+                 * - Initial assessment fails.
+                 * - Full assessment is performed and fails.
+                 * - Hardship review also fails.
+                 * - Expected outcome: GRANTED_FAILED_MEANS_TEST.
+                 */
+                Arguments.of(
+                        CaseType.INDICTABLE,
+                        MagCourtOutcome.APPEAL_TO_CC,
+                        CurrentStatus.COMPLETE,
+                        CurrentStatus.COMPLETE,
                         InitAssessmentResult.FAIL.getResult(),
                         FullAssessmentResult.FAIL.getResult(),
-                        ReviewResult.FAIL)
-        );
-    }
-
-    @Test
-    void givenInProgressFullAssessment_whenGetRepDecisionIsInvoked_nullDecisionIsReturned() {
-        CrownCourtDTO requestDTO = TestModelDataBuilder.getCrownCourtDTO();
-        setUpFinAssessment(requestDTO, CurrentStatus.COMPLETE, CurrentStatus.IN_PROGRESS,
-                           InitAssessmentResult.FAIL.getResult(), FullAssessmentResult.FAIL.getResult(),
-                           ReviewResult.FAIL
+                        ReviewResult.FAIL,
+                        Constants.GRANTED_FAILED_MEANS_TEST
+                ),
+                /* Scenario 3: INDICTABLE case with MagCourtOutcome of APPEAL_TO_CC.
+                 * - Initial assessment fails.
+                 * - Full assessment is performed and fails.
+                 * - Hardship review passes.
+                 * - Expected outcome: GRANTED_PASSED_MEANS_TEST.
+                 */
+                Arguments.of(
+                        CaseType.INDICTABLE,
+                        MagCourtOutcome.APPEAL_TO_CC,
+                        CurrentStatus.COMPLETE,
+                        CurrentStatus.COMPLETE,
+                        InitAssessmentResult.FAIL.getResult(),
+                        FullAssessmentResult.FAIL.getResult(),
+                        ReviewResult.PASS,
+                        Constants.GRANTED_PASSED_MEANS_TEST
+                ),
+                /*
+                 * Scenario 4: APPEAL_CC case with MagCourtOutcome of COMMITTED_FOR_TRIAL.
+                 * - Initial assessment passes.
+                 * - Full assessment is in progress.
+                 * - Hardship review passes.
```

Comment:
I know this is an existing test scenario that you've refactored, but I'm just wondering generally how we can have a hardship review result when the full means assessment hasn't been completed.

Reply:
You're right, this makes no sense.